lib/Folder/Folder.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\GroupFolders\Folder;
+
+use OC\Files\Cache\CacheEntry;
+
+class Folder {
```

Comment:
Avoid using the same name as core classes, it’s confusing when the class is imported with a `use` statement.

Reply:
But `Groupfolder` is also not a good name. Would `InternalFolder` be better as it is only used as an internal type between different methods?

Reply:
As said below this should be in Db and be an entity with a mapper ideally.
I do think Groupfolder is closer to what it is than Folder.
Could be FolderDefinition or FolderSettings if that makes sense?